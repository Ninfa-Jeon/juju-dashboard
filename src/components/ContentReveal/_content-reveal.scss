.content-reveal {
  & + & {
    border-top: 1px solid #ccc;
  }

  &__header {
    cursor: pointer;
    display: flex;
    font-weight: 400;
    padding: 0.5rem;
    place-items: center;
    width: 100%;

    i {
      display: inline-block;
      margin-right: 0.5rem;
    }
  }

  &__title {
    display: flex;

    .chip-group {
      display: flex;
      margin-bottom: 0 !important;
      padding-bottom: 0 !important;
      padding-left: 1rem;
    }
  }

  &__content {
    max-height: 100%;
    overflow-x: scroll;
    padding: 1rem 0.5rem 0;

    &[aria-hidden="true"] {
      max-height: 0;
      overflow: hidden;
      padding-bottom: 0;
      padding-top: 0;
    }
  }
}
