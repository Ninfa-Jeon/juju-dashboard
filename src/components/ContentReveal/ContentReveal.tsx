import { useState } from "react";

import "./_content-reveal.scss";

type Props = {
  title: string | JSX.Element;
  children: JSX.Element;
  showContent: boolean;
};

export default function ContentReveal({ title, showContent, children }: Props) {
  const [isOpen, setIsOpen] = useState(showContent);

  function handleToggle() {
    setIsOpen(!isOpen);
  }

  const iconName = isOpen ? "minus" : "plus";

  return (
    <div className="content-reveal">
      <div
        className="content-reveal__header"
        onClick={() => handleToggle()}
        onKeyDown={() => handleToggle()}
        role="button"
        tabIndex={0}
      >
        <i className={`p-icon--${iconName}`}></i>
        <div className="content-reveal__title">{title}</div>
      </div>
      <div className="content-reveal__content" aria-hidden={!isOpen}>
        {children}
      </div>
    </div>
  );
}
