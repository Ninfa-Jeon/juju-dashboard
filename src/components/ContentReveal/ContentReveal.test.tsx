import { mount } from "enzyme";

import ContentReveal from "./ContentReveal";

describe("Content Reveal", () => {
  it("should show children as content", () => {
    const wrapper = mount(
      <ContentReveal title="Foo bar" openByDefault={false}>
        <p>Banner text</p>
      </ContentReveal>
    );
    expect(wrapper.find("p").text()).toStrictEqual("Banner text");
  });

  it("should show correct title if only text", () => {
    const wrapper = mount(
      <ContentReveal title="Foo bar" openByDefault={false}>
        <p>Banner text</p>
      </ContentReveal>
    );
    expect(wrapper.find(".content-reveal__title").text()).toStrictEqual(
      "Foo bar"
    );
  });

  it("should show correct title if only text and JSX", () => {
    const Component = () => <div className="component">component content</div>;
    const getComponent = () => {
      return Component;
    };
    const wrapper = mount(
      <ContentReveal title={`Foo bar ${getComponent()}`} openByDefault={false}>
        <p>Banner text</p>
      </ContentReveal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should open by default if set", () => {
    const wrapper = mount(
      <ContentReveal title="Foo bar" openByDefault={true}>
        <p>Banner text</p>
      </ContentReveal>
    );
    expect(
      wrapper.find(".content-reveal__content").prop("aria-hidden")
    ).toStrictEqual(false);
  });
});
