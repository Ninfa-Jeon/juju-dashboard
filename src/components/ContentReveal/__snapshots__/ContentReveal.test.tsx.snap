// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Reveal should show correct title if only text and JSX 1`] = `
<ContentReveal
  openByDefault={false}
  title="Foo bar () => /*#__PURE__*/(0, _jsxDevRuntime.jsxDEV)(\\"div\\", {
      className: \\"component\\",
      children: \\"component content\\"
    }, void 0, false, {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 29
    }, void 0)"
>
  <div
    className="content-reveal"
  >
    <div
      className="content-reveal__header"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <i
        className="p-icon--plus"
      />
      <div
        className="content-reveal__title"
      >
        Foo bar () =&gt; /*#__PURE__*/(0, _jsxDevRuntime.jsxDEV)("div", {
      className: "component",
      children: "component content"
    }, void 0, false, {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 29
    }, void 0)
      </div>
    </div>
    <div
      aria-hidden={true}
      className="content-reveal__content"
    >
      <p>
        Banner text
      </p>
    </div>
  </div>
</ContentReveal>
`;
